src/shortcuts/deleteEmptyThoughtOrOutdent.js

```diff
@@ -60,6 +63,33 @@ const exec = (dispatch, getState) => {
   dispatch({ type: 'deleteEmptyThought' })
 }
 
+// eslint-disable-next-line jsdoc/require-jsdoc
+const canExecuteOutdent = getState => {
+  const state = getState()
+  const { cursor } = state
+
+  const parentContext = contextOf(pathToContext(cursor))
+  const children = getThoughts(state, parentContext)
+
+  const offset = window.getSelection().focusOffset
+
+  // eslint-disable-next-line
+  const filteredChildren = () => children.filter(child => state.showHiddenThoughts ||
+    // exclude meta thoughts when showHiddenThoughts is off
+    (!isFunction(child.value) && !hasChild(state, unroot(parentContext.concat(child.value)), '=hidden'))
+  )
+  return isDocumentEditable() && headValue(cursor).length !== 0 && offset === 0 && filteredChildren().length === 1
+}
+
+// eslint-disable-next-line jsdoc/require-jsdoc
+const canExecute = getState => canExecuteOutdent(getState) || canExecuteDeleteEmptyThought(getState)
+
+// eslint-disable-next-line jsdoc/require-jsdoc
+const exec = (dispatch, getState) => {
+  if (canExecuteOutdent(getState)) dispatch({ type: 'outdent' })
+  else deleteEmptyThought(dispatch, getState)
```

Comment:
This deserves a comment indicating that `canExecuteDeletEmptyThought` has already been checked in `canExecute`.

Reply:
It can also be called in the conventional thunk action-creator form:

```js
dispatch(deleteEmptyThought)
```